### ISSUE TITLE
sync vs async

### ISSUE THREAD

**Issue by ralphtheninja** [nikhilk/node-tensorflow#17]
So the `.run()` operation is synchronous. If this can take a lot of time it doesn't really play nice with the node way of doing things.

```js
// Run to evaluate and retrieve the value of the 'result' op.
let outputs = session.run(/* inputs */ null,
                          /* outputs */ [ 'result' ],
                          /* targets */ null);

// Should print out '42'
console.log(outputs.result.toValue());
```

Is it possible to make operations like this async using `node-ffi`?

**Comment by ralphtheninja** [comment]
Ideally `session.run()` should take a callback as last argument _or_ it can return a `Promise` which resolves to `result`.

**Comment by nikhilk** [comment]
Agree; need to see if we can make this work with ffi, and add a runAsync that returns a Promise.

If we can't with just ffi it may be an interesting feature request for the TensorFlow C API.

**Comment by nikhilk** [comment]
This seems simpler than I thought it might have been with what ffi supports -- for reference: https://github.com/node-ffi/node-ffi/wiki/Node-FFI-Tutorial#async-library-calls
